lib/core/src/Cardano/Wallet/DaedalusIPC.hs

```diff
@@ -214,7 +214,7 @@ readMessage :: Handle -> IO BL.ByteString
 readMessage = if isWindows then windowsReadMessage else posixReadMessage
 
 isWindows :: Bool
-isWindows = arch == "windows"
+isWindows = os == "windows"
```

Comment:
Just a thought, but perhaps we can replace this `Bool` check with an enumeration that includes just the platforms we care about. Perhaps like this:

```hs
data Architecture = Windows | Linux | MacOS
```
Then we can have:
```hs
 readMessage = \case
    Windows -> windowsReadMessage
    Linux -> posixReadMessage
    MacOS -> macOsReadMessage
```

Reply:
For what it's worth here, it's really a local check in one place. I don't think we need much more than this.